sdk/search/azure-search/src/main/java/com/azure/search/models/Suggester.java

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package com.azure.search.models;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.List;
+
+/**
+ * Defines how the Suggest API should apply to a group of fields in the index.
+ */
+@Fluent
+public final class Suggester {
+    /*
+     * The name of the suggester.
+     */
+    @JsonProperty(value = "name", required = true)
+    private String name;
+
+    /*
+     * A value indicating the capabilities of the suggester.
+     */
+    @JsonProperty(value = "searchMode", required = true)
+    private String searchMode;
+
+    /*
+     * The list of field names to which the suggester applies. Each field must
+     * be searchable.
+     */
+    @JsonProperty(value = "sourceFields", required = true)
+    private List<String> sourceFields;
+
+    /**
+     * Creates an instance of Suggester class.
+     */
+    public Suggester() {
+        searchMode = "analyzingInfixMatching";
+    }
+
+    /**
+     * Get the name property: The name of the suggester.
+     *
+     * @return the name value.
+     */
+    public String getName() {
+        return this.name;
+    }
+
+    /**
+     * Set the name property: The name of the suggester.
+     *
+     * @param name the name value to set.
+     * @return the Suggester object itself.
+     */
+    public Suggester setName(String name) {
+        this.name = name;
+        return this;
+    }
+
+    /**
+     * Get the searchMode property: A value indicating the capabilities of the
+     * suggester.
+     *
+     * @return the searchMode value.
+     */
+    public String getSearchMode() {
```

Comment:
This looks like an unfortunate difference in behavior between the C# and Java code generators. In C#, when an enum has only one value, the property is static and immutable, so customers can ignore it and can't hurt themselves by changing it. It should be the same in Java IMO.

One way or another, we should find a way to remove `setSearchMode` and make `getSearchMode` static before preview.

Reply:
The same issue exists for `SynonymMap` -- `getFormat` and `setFormat`.

Reply:
As those are generated model classes, this change need to be done in the autorest config and followed by code re-generation

Reply:
Or we fix it in AutoRest, but either way it happens after the PR is merged.